pkg/client/rebase.go

```diff
@@ -56,7 +62,18 @@ func (c *Client) Rebase(ctx context.Context, opts RebaseOptions) error {
 		return errors.Wrapf(err, "invalid image name '%s'", opts.RepoName)
 	}
 
-	appImage, err := c.imageFetcher.Fetch(ctx, opts.RepoName, image.FetchOptions{Daemon: !opts.Publish, PullPolicy: opts.PullPolicy})
+	appImageName := opts.RepoName
+
+	if len(opts.Tags) > 0 {
+		parts := strings.SplitN(appImageName, tagDelim, 2)
+		if len(parts) == 2 {
+			appImageName = fmt.Sprintf("%s:%s", parts[0], opts.Tags[0])
+		} else {
+			appImageName = fmt.Sprintf("%s:%s", appImageName, opts.Tags[0])
+		}
+	}
+
+	appImage, err := c.imageFetcher.Fetch(ctx, appImageName, image.FetchOptions{Daemon: !opts.Publish, PullPolicy: opts.PullPolicy})
```

Comment:
I think this is still going to replace the target image in the registry, we are just specifying a different target image now. I think we want to read in something like `opts.PreviousImage` and then (looking at the signature for `rebaser.Rebase`):

```
Rebase(workingImage imgutil.Image, newBaseImage imgutil.Image, outputImageRef string, additionalNames []string)
```

`workingImage` will be based off of `opts.PreviousImage`
`newBaseImage` is derived as before (from the working image)
`outputImageRef` comes from the first argument to `pack rebase` (not `opts.PreviousImage`)
`additionalNames` is nil as before

Reply:
(We probably also want some kind of test here)

Reply:
@natalieparellano thanks for the suggestions, I will commit the desired changes.